src/lib/vocab/entity/entityMutationHelpers.ts

```diff
@@ -85,3 +85,38 @@ export const updateTieCaches = (
 		destTies!.mutate(($v) => $v.push($tie));
 	}
 };
+
+export const evictTieCachesForEntity = (
+	{sourceTiesByDestEntityId, destTiesBySourceEntityId}: WritableUi,
+	entity_id: number,
+): void => {
+	const sources = sourceTiesByDestEntityId.get().value.get(entity_id);
+	if (sources) {
+		for (const source of sources.get().value) {
+			const ties = destTiesBySourceEntityId.get().value.get(source.source_id);
+			if (ties) {
+				ties.mutate(($ties) => {
+					for (let i = $ties.length - 1; i >= 0; i--) {
+						if ($ties[i].dest_id === entity_id) $ties.splice(i, 1);
+					}
+				});
+			}
+		}
+		sourceTiesByDestEntityId.mutate(($v) => $v.delete(entity_id));
+	}
+
+	const destinations = destTiesBySourceEntityId.get().value.get(entity_id);
+	if (destinations) {
+		for (const dest of destinations.get().value) {
+			const ties = sourceTiesByDestEntityId.get().value.get(dest.source_id);
```

Comment:
I think this is right? not sure tho

```suggestion
			const ties = sourceTiesByDestEntityId.get().value.get(dest.dest_id);
```

Reply:
yes, you're totally right!